docs/reference/shopifyApi.md

```diff
@@ -0,0 +1,189 @@
+
+### apiVersion
+
+`ApiVersion` | Defaults to `LATEST_API_VERSION`
+
+API version your app will be querying. E.g. `ApiVersion.January20`.
+
+### isEmbeddedApp
+
+`boolean` | Defaults to `true`
+
+Whether your app will run within the Shopify Admin. Learn more about embedded apps with [`App Bridge`](https://shopify.dev/apps/tools/app-bridge/getting-started/app-setup).
+
+### isPrivateApp
+
+`boolean` | Defaults to `false`
+
+Whether you are building a private app for a store.
+
+### userAgentPrefix
+
+`string` | Defaults to `undefined`
+
+Any prefix you wish to include in the `User-Agent` for requests made by the library.
+
+### privateAppStorefrontAccessToken
+
+`string` | Defaults to `undefined`
+
+Fixed Storefront API access token for private apps.
+
+### customShopDomains
+
+`(RegExp | string)[]` | Defaults to `undefined`
+
+Use this if you need to allow values other than `myshopify.com`.
+
+### billing
+
+`BillingConfig` | Defaults to `undefined`
+
+Billing configurations. [See documentation](../usage/billing.md) for full description.
+
+### restResources
+
+`ShopifyRestResources`
+
+Mounts the given REST resources onto the object. Must use the same version as `apiVersion`. Learn more about [using REST resources](../usage/rest-resources.md).
+
+### logger
+
+`LoggerConfig`
+
+Tweaks the behaviour of the package's internal logging to make it easier to debug applications.
+
+#### log
+
+`() => Promise<void>`
+
+Async callback function used for logging, which takes in a `LogSeverity` value and a formatted `message`. Defaults to using `console` calls matching the severity parameter.
+
+#### level
+
+`LogSeverity` | Defaults to `LogSeverity.Info`
+
+Minimum severity for which to trigger the log function.
+
+#### httpRequests
+
+`boolean` | Defaults to `false`
+
+Whether to log **ALL** HTTP requests made by the package. Logs the requests at the `Debug` level.
+
+#### timestamps
+
+`boolean` | Defaults to `false`
+
+Whether to add the current timestamp to every log call.
+
+## Return
+
+This function returns an object containing the following properties:
+
+### config
+
+The options used to set up the object, containing the parameters of this function.
+
+### [auth](./auth/README.md)
+
+Object containing functions to authenticate with Shopify APIs.
+
+### [clients](./clients/README.md)
+
+Object containing clients to access Shopify APIs.
+
+### [session](./session/README.md)
+
+Object containing functions to manage Shopify sessions.
+
+### [webhooks](./webhooks/README.md)
+
+Object containing functions to handle Shopify webhooks.
+
+### [billing](./billing/README.md)
+
+Object containing functions to enable apps to bill merchants.
+
+### [utils](./utils/README.md)
+
+Object containing general functions to help build apps.
+
+### rest
+
+Object containing OO representations of the Admin REST API.
+See the [API reference documentation](https://shopify.dev/api/admin-rest) for details.
+
+[Back to reference index](./README.md)
```

Comment:
I really like this link back to the reference index. Could we add it to the other sub reference pages?